app/models/spree/slide.rb

```diff
@@ -33,7 +33,7 @@ def slide_name
   end
 
   def slide_link
-    link_url.blank? && product.present? ? product : link_url
+    link_url.blank? && product.present? ? "/products/#{product.slug}?taxon_id=#{product.id}" : link_url
```

Comment:
I think we have helper for product url.. something like `product_url(:slug)`

taxon_id cannot be product.id..
You can instead discard taxon_id and frontend is able to auto figure that out..